app/src/org/commcare/dalvik/geo/HereFunctionHandler.java

```diff
@@ -0,0 +1,149 @@
+                    (provider.equals(LocationManager.NETWORK_PROVIDER) && ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED)) {
+                // This is non-null if the calling activity pauses.
+                if (location == null) {
+                    Location lastKnownLocation = mLocationManager.getLastKnownLocation(provider);
+                    this.location = toGeoPointData(lastKnownLocation);
+                    Log.i("HereFunctionHandler", "last known location: " + this.location.getDisplayText());
+                }
+
+                // Looper is necessary because requestLocationUpdates is called inside an AsyncTask (EntityLoaderTask).
+                // What values for minTime and minDistance?
+                mLocationManager.requestLocationUpdates(provider, 0, 0, this, Looper.getMainLooper());
+            }
+        }
+
+        locationUpdatesRequested = true;
+    }
+
+    // Clients must call this when done using handler.
+    public void stopLocationUpdates() {
+        // stops the GPS. Note that this will turn off the GPS if the screen goes to sleep.
+        if (ContextCompat.checkSelfPermission(this.context, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED ||
+                ContextCompat.checkSelfPermission(this.context, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
+            mLocationManager.removeUpdates(this);
+        }
+
+        locationUpdatesRequested = false;
+    }
+
+    @Override
+    public void onLocationChanged(Location location) {
+        // Do we need to check the accuracy of the location?
+        this.location = toGeoPointData(location);
+        Log.i("HereFunctionHandler", "location has been set to " + this.location.getDisplayText());
+        // Trigger refreshing of entity list.
+        if (entitySelectActivity != null) {
+            entitySelectActivity.loadEntities();
+        }
+    }
+
+    public static GeoPointData toGeoPointData(Location location) {
+        return new GeoPointData(new double[]{
+                location.getLatitude(),
+                location.getLongitude(),
+                location.getAltitude(),
+                (double) location.getAccuracy()
+        });
+    }
+
+    @Override
+    public void onProviderDisabled(String provider) {}
+
+
+    @Override
+    public void onProviderEnabled(String provider) {}
+
+
+    @Override
+    public void onStatusChanged(String provider, int status, Bundle extras) {}
+
+    // Methods for implementing IFunctionHandler
+
+    public String getName() {
+        return HERE_NAME;
+    }
+
+    public Vector getPrototypes() {
+        Vector p = new Vector();
+        p.addElement(new Class[0]);
+        return p;
+    }
+
+    public boolean rawArgs() {
+        return false;
+    }
+
+    public boolean realTime() {
```

Comment:
I recently removed this from the IFunctionHandler interface, so you can remove this implementation, since it will never be called